docs/references/nextjs/clerk-middleware.mdx

```diff
@@ -295,4 +295,20 @@ The `clerkMiddleware()` function accepts an optional object. The following optio
 | `signInUrl?`                       | `string`             | An alternative sign in URL.                                                                                                                                                                                                                                                                                                                     |
 | `signUpUrl?`                       | `string`             | An alternative sign up URL.                                                                                                                                                                                                                                                                                                                     |
 | `publishableKey` | `string` | The Clerk publishable key for your instance. This can be found in your Clerk Dashboard on the **[API Keys](https://dashboard.clerk.com/last-active?path=api-keys)** page.|
-| `secretKey?` | `string` | The Clerk secret key for your instance. This can be found in your Clerk Dashboard on the **[API Keys](https://dashboard.clerk.com/last-active?path=api-keys)** page.  |
+| `secretKey?` | `string` | The Clerk secret key for your instance. This can be found in your Clerk Dashboard on the **[API Keys](https://dashboard.clerk.com/last-active?path=api-keys)** page. The `CLERK_ENCRYPTION_KEY` environment variable must be  when providing `secretKey` as an option, refer to [Dynamic keys](#dynamic-keys). |
+
+### Dynamic keys
+
+The following options, known as "Dynamic Keys," are shared to the Next.js application server through `clerkMiddleware`, enabling access by server-side helpers like [`auth()`](/docs/references/nextjs/auth):
+- `signUpUrl`
+- `signInUrl`
+- `secretKey`
+- `publishableKey`
+
+Dynamic keys are encrypted and shared during request time using a [AES encryption algorithm](https://en.wikipedia.org/wiki/Advanced_Encryption_Standard). When providing a `secretKey`, the `CLERK_ENCRYPTION_KEY` environment variable is mandatory and used as the encryption key. If no `secretKey` is provided to `clerkMiddleware`, the encryption key defaults to `CLERK_SECRET_KEY`.
+
+When providing `CLERK_ENCRYPTION_KEY`, a 128-bit, pseudorandom value should be used.
```

Comment:
We can suggest the usage of `openssl` to easily generate a key (need to wrap the command in back ticks, breaks suggestion formatting):

```suggestion
When providing `CLERK_ENCRYPTION_KEY`, it is recommended to use a 32-byte (256-bit), pseudorandom value. You can use `openssl` to generate a key:

$ openssl rand --hex 32
```